PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/survey/CustomerSurvey.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.intellij.common.survey;
+
+import com.intellij.ide.plugins.PluginManagerCore;
+import com.intellij.openapi.application.ApplicationInfo;
+import com.intellij.openapi.extensions.PluginDescriptor;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.util.IconLoader;
+import com.microsoft.azure.toolkit.lib.common.telemetry.AzureTelemeter;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.commons.lang.StringUtils;
+
+import javax.swing.*;
+import java.util.Optional;
+
+@Getter
+@AllArgsConstructor
+public enum CustomerSurvey implements ICustomerSurvey {
+
+    AZURE_SDK(Constants.AZURE_SDK_SURVEY_LINK, Constants.AZURE_SDK_SURVEY_DESCRIPTION, Constants.AZURE_ICON) {
+        @Override
+        public String getLink() {
+            return String.format(super.getLink(), getInstallationId());
+        }
+    },
+    AZURE_CLIENT_SDK(Constants.AZURE_CLIENT_SDK_SURVEY_LINK, Constants.AZURE_CLIENT_SDK_SURVEY_DESCRIPTION, Constants.AZURE_ICON) {
+        @Override
+        public String getLink() {
+            return String.format(super.getLink(), getInstallationId());
+        }
+    },
+    AZURE_MGMT_SDK(Constants.AZURE_MGMT_SDK_SURVEY_LINK, Constants.AZURE_MGMT_SDK_SURVEY_DESCRIPTION, Constants.AZURE_ICON) {
+        @Override
+        public String getLink() {
+            return String.format(super.getLink(), getInstallationId());
+        }
```

Comment:
how about making the getLink() implementation in their super interface as default ?

Reply:
no, as get link differs for different survey, only sdk survey shares same strategy, i don't think this should be make as the default implementation